common/presenters/framework-section-to-panel-list.js

```diff
@@ -51,42 +53,46 @@ function frameworkSectionToPanelList({
     const sectionStatus = find(personEscortRecord?.meta?.section_progress, {
       key: section.key,
     })?.status
+    const panels = tagList
+      // Filter tags to just this section
+      .filter(tag => tag.section === section.key)
+      // Return a meta list component for each tag
+      .map(tag => {
+        // Loop over each item to append question description
+        const flagsWithQuestionDescription = (flagsMap[tag.text] || []).map(
+          response => {
+            const { description } = questions[response.question.key]
+
+            return {
+              ...response,
+              question: {
+                ...response.question,
+                description,
+              },
+            }
+          }
+        )
+        const metaList = frameworkResponseToMetaListComponent(
+          flagsWithQuestionDescription
+        )
+
+        return {
+          // Remove link so that when displayed in the main
+          // body the tag isn't linked
+          tag: omit(tag, ['href']),
+          attributes: { id: kebabCase(tag.text) },
+          html: componentService.getComponent('appMetaList', metaList),
+        }
+      })
 
     const output = {
       key: section.key,
       name: section.name,
       url: `${personEscortRecordUrl}/${section.key}`,
       isCompleted: sectionStatus === 'completed',
-      panels: tagList
-        // Filter tags to just this section
-        .filter(tag => tag.section === section.key)
-        // Return a meta list component for each tag
-        .map(tag => {
-          // Loop over each item to append question description
-          const flagsWithQuestionDescription = (flagsMap[tag.text] || []).map(
-            response => {
-              const { description } = questions[response.question.key]
-
-              return {
-                ...response,
-                question: {
-                  ...response.question,
-                  description,
-                },
-              }
-            }
-          )
-
-          return {
-            // Remove link so that when displayed in the main
-            // body the tag isn't linked
-            tag: omit(tag, ['href']),
-            attributes: { id: kebabCase(tag.text) },
-            metaList: frameworkResponseToMetaListComponent(
-              flagsWithQuestionDescription
-            ),
-          }
-        }),
+      count: panels.length,
```

Comment:
I think there is still a large amount of improvement around this presenter but didn't want to tackle it all at once.

I've made [some improvements](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/1070) to the way the person escort record gets transformer that will hopefully allow this to be simplified.

Reply:
Where is `count` actually being used?

Reply:
It's used in [the macro](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/e83f3e9812c497769d2243c37d320cac86ebbc78/app/move/views/view.njk#L262) in the view file. The main reason for this change is so that this same structure can be passed to that macro.

I'm going to look to also simplify this once #1070 is in.